src/nspawn/nspawn.c

```diff
@@ -195,6 +194,9 @@ static char **arg_parameters = NULL;
 static const char *arg_container_service_name = "systemd-nspawn";
 static bool arg_notify_ready = false;
 static bool arg_use_cgns = true;
+static bool arg_share_ns_ipc = false;
+static bool arg_share_ns_pid = false;
+static bool arg_share_ns_uts = false;
```

Comment:
I think we should just make this a mask, that reuses the kernel's flags:

```c
static unsigned long arg_share_ns = CLONE_IPC|CLONE_PID|CLONE_UTS;
````

And the three env vars and the cmdline arg should then just mask/add to this flags field.